plugins/techdocs/src/reader/components/TechDocsPageHeader.tsx

```diff
@@ -51,19 +52,39 @@ export const TechDocsPageHeader = ({
     spec: { owner, lifecycle },
   } = entityMetadataValues || { spec: {} };
 
-  const componentLink = `/catalog/${kind}/${name}`;
+  const componentLink = useRouteRef(entityRouteRef);
+
+  let ownerEntity;
+  if (owner) {
+    ownerEntity = parseEntityName(owner, { defaultKind: 'group' });
+  }
 
   const labels = (
     <>
       <HeaderLabel
         label="Component"
         value={
-          <Link style={{ color: '#fff' }} to={componentLink}>
+          <Link style={{ color: '#fff' }} to={componentLink(entityId)}>
             {name}
           </Link>
         }
       />
-      {owner ? <HeaderLabel label="Site Owner" value={owner} /> : null}
+      {owner ? (
+        <HeaderLabel
+          label="Site Owner"
```

Comment:
Since we're here, I've always thought this one was a bit odd.  A "site" infers a "web site". That is great if the entity is a `kind:component` + `spec.type:website`, but not if it's a `kind:component` + `spec.type:library` which might have a techdocs annotation to describe how the library works, but those documents might not actually represent a (web)site.

Might this header description also work as "Doc Owner" or "Docs Owner"? open to other opinions on it?

Reply:
Good point @adamdmharvey what if we just stick to "Owner" instead? Do we need to give more context than that do you think?

Reply:
even better !  `Owner: @emmaindal`